File: python/3.py
from itertools import combinations

def target_sum(nums, target):
    for r in range(1, len(nums) + 1):
        for subset in combinations(nums, r):
            if sum(subset) == target:
                return "yes"
    return "no"

tedad = int(input())
nums = list(map(int, input().split()))
target = int(input())
result = target_sum(nums, target)
print(result)

File: C/3.cpp
#include <iostream>
#include <vector>
using namespace std;

bool targetSum(const vector<int>& nums, int target) {
    int n = nums.size();
    // بررسی تمام زیرمجموعه‌ها
    for (int i = 1; i < (1 << n); ++i) {
        int sum = 0;
        for (int j = 0; j < n; ++j) {
            if (i & (1 << j)) {
                sum += nums[j];
            }
        }
        if (sum == target) {
            return true;
        }
    }
    return false;
}

int main() {
    int n;
    cin >> n;

    vector<int> nums(n);
    for (int i = 0; i < n; ++i) {
        cin >> nums[i];
    }

    int target;
    cin >> target;

    if (targetSum(nums, target))
        cout << "yes\n";
    else
        cout << "no\n";
    
    return 0;
}
